src/accelerate/hooks.py

```diff
@@ -179,6 +181,10 @@ def remove_hook_from_module(module: nn.Module):
         module.forward = module._old_forward
         delattr(module, "_old_forward")
 
+    if recurse and len(list(module.children())) > 1:
```

Comment:
Not sure the check is useful (should be >0 but the loop will just be empty)
```suggestion
    if recurse:
```

Reply:
Nice catch! Yes indeed ;)